src/test/java/uk/gov/companieshouse/presenter/account/controller/PresenterAccountControllerTest.java

```diff
@@ -1,27 +1,181 @@
+        var presenterDetails = createPresenterDetails(
+            USER_ID,
+            EMAIL,
+            FIRST_NAME,
+            LAST_NAME,
+            PREMISES,
+            FIRST_LINE,
+            "",
+            COUNTY,
+            COUNTRY,
+            POSTCODE);
+        var response = presenterAccountController.createPresenterAccount(presenterDetails);
+        var header = response.getHeaders().getFirst("Location");
+        assertThat(response.getStatusCode(), is(HttpStatus.CREATED));
+        assertTrue(header.contains(PRESENTER_ACCOUNT));
+        assertFalse(header.contains(PRENETER_ACCOUNT_NULL));
+        assertThat(matchesPattern(PRESENTER_ACCOUNT_REGEX).matches(header), is(true));
+    }
+
+    @Test
+    @DisplayName("Return 201 when valid presenter details with county is missing is submitted")
+    void testCreatePresenterAccountMissingCountySuccessResponse() {
+        var presenterDetails = createPresenterDetails(
+            USER_ID,
+            EMAIL,
+            FIRST_NAME,
+            LAST_NAME,
+            PREMISES,
+            FIRST_LINE,
+            SECOND_LINE,
+            "",
+            COUNTRY,
+            POSTCODE);
+        var response = presenterAccountController.createPresenterAccount(presenterDetails);
+        var header = response.getHeaders().getFirst("Location");
+        assertThat(response.getStatusCode(), is(HttpStatus.CREATED));
+        assertTrue(header.contains(PRESENTER_ACCOUNT));
+        assertFalse(header.contains(PRENETER_ACCOUNT_NULL));
+        assertThat(matchesPattern(PRESENTER_ACCOUNT_REGEX).matches(header), is(true));
+    }
+
+    // NOT TESTING MISSING NON-OPTIONAL PARAMETER AS THEY ARE CATCH BY MODEL.
+    // Will test ValidationException handling
+    @Test
+    @DisplayName("ValidationException thrown return 400")
+    void testThrowValidationExcetionBadRequestResponse() {
+        ValidationException e = new ValidationException();
+
+        ResponseEntity<?> response = presenterAccountController.validationException(e);
+
+        assertThat(response.getStatusCode(), is(HttpStatus.BAD_REQUEST));
+    }
+
+    @Test
+    @DisplayName("HttpMessageNotReadableException thrown return 400")
+    void testThrowHttpMessageNotReadableExceptionBadRequestResponse() {
+        HttpMessageNotReadableException e = mock(HttpMessageNotReadableException.class);
+
+        ResponseEntity<?> response = presenterAccountController.httpMessageNotReadableException(e);
+
+        assertThat(response.getStatusCode(), is(HttpStatus.BAD_REQUEST));
+    }
+
+    @Test
+    @DisplayName("Exception thrown returns 500")
+    void testThrowExcetionInternalError() {
+        Exception e = new Exception();
+
+        ResponseEntity<?> response = presenterAccountController.exceptionHandler(e);
+
+        assertThat(response.getStatusCode(), is(HttpStatus.INTERNAL_SERVER_ERROR));
+    }
+
+    private PresenterRequest createPresenterDetails(
```

Comment:
You don't need to create all this detail in the presenter request for the controller test. The controller never checks any of the data, that's done by the service which is mocked